suprb/optimizer/rule/ns/novelty_calculation.py

```diff
@@ -0,0 +1,100 @@
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+from scipy.spatial.distance import hamming
+from .novelty_search_type import NoveltySearchType
+from .archive import Archive
+
+
+class NoveltyCalculation(BaseComponent, metaclass=ABCMeta):
+    def __init__(self, novelty_search_type: NoveltySearchType, archive: Archive, k_neighbor: int):
+        self.novelty_search_type = novelty_search_type
+        self.archive = archive
+        self.k_neighbor = k_neighbor
```

Comment:
I would prefer the correct spelling here. what is the python convention for this?

Reply:
Sadly this is the scipy convention https://scikit-learn.org/stable/modules/neighbors.html